pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/IndexHandler.java

```diff
@@ -39,4 +39,12 @@ void updateIndices(SegmentDirectory.Writer segmentWriter, IndexCreatorProvider i
    */
   boolean needUpdateIndices(SegmentDirectory.Reader segmentReader)
       throws Exception;
+
+  /**
+   * Performs any cleanup actions required after the indexes have been updated.
+   * Should be called only after all IndexHandlers have run.
+   */
+  default void postUpdateIndicesCleanup(SegmentDirectory.Writer segmentWriter)
```

Comment:
Hmmm not sure if I follow this. Seems a bit confusing. If we want this to be invoked only after all indexHandlers have run, then I don't think this API should exist on `IndexHandler` interface.

Reply:
Nvm...I just saw the change in `SegmentPreprocessor` and I think this makes sense.